a3p-integration/proposals/z:acceptance/vaults.test.js

```diff
@@ -0,0 +1,201 @@
+import test from 'ava';
+import {
+  agops,
+  bankSend,
+  getUser,
+  openVault,
+  adjustVault,
+  closeVault,
+  getISTBalance,
+  queryVstorage,
+  ATOM_DENOM,
+  USER1ADDR,
+  waitForBlock,
+} from '@agoric/synthetic-chain';
+
+const getFormattedVaultData = vaultData => {
+  const { values } = JSON.parse(vaultData.value);
+  const { body } = JSON.parse(values[0]);
+  const cleanedBody = JSON.parse(body.slice(1));
+  return cleanedBody;
+};
+
+test.serial('attempt to open vaults under the minimum amount', async t => {
+  const activeVaultsBefore = await agops.vaults('list', '--from', USER1ADDR);
+  await bankSend(USER1ADDR, `20000000${ATOM_DENOM}`);
+
+  const mint = '3.0';
+  const collateral = '5.0';
+  await t.throwsAsync(() => openVault(USER1ADDR, mint, collateral));
+
+  const activeVaultsAfter = await agops.vaults('list', '--from', USER1ADDR);
+
+  t.true(
+    activeVaultsAfter.length === activeVaultsBefore.length,
+    'The number of active vaults should remain the same.',
+  );
+});
+
+test.serial('open new vault', async t => {
+  const istBalanceBefore = await getISTBalance(USER1ADDR);
+  const activeVaultsBefore = await agops.vaults('list', '--from', USER1ADDR);
+
+  await bankSend(USER1ADDR, `20000000${ATOM_DENOM}`);
+
+  const mint = '5.0';
+  const collateral = '10.0';
+  await openVault(USER1ADDR, mint, collateral);
+  await waitForBlock();
+
+  const istBalanceAfter = await getISTBalance(USER1ADDR);
+  const activeVaultsAfter = await agops.vaults('list', '--from', USER1ADDR);
+
+  t.true(
+    istBalanceBefore + 5 === istBalanceAfter,
+    'The IST balance should increase by the minted amount',
+  );
+  t.true(
+    activeVaultsAfter.length > activeVaultsBefore.length,
```

Comment:
Thank you for bringing this to my attention.
Since we are opening a single new Vault, the difference should always be 1 Vault.
Change addressed on commit [1894140](https://github.com/Agoric/agoric-sdk/pull/10123/commits/1894140c246d6935f41805d4da3b22cf592abdc2)